### PR TITLE
Allow platforms at local paths and arbitrary git repositories

### LINKED ISSUE #26: Depend on tagged aemm-android, not HEAD
aemmobile is currently dependent on the HEAD revision of aemm-android, when it should be dependent on the latest tagged release. Retrieving the latest tagged release should be possible with GitHub's latest release API:

https://developer.github.com/v3/repos/releases/#get-the-latest-release

### LINKED ISSUE #33: Support local directory and git url platform specs
Like the [cordova platform command](https://cordova.apache.org/docs/en/latest/reference/cordova-cli/index.html#cordova-platform-command) it would be useful to specify a local directory or git url when adding a platform using `aemm platform add`.  E.g.

```
aemm platform add ../localBranchOfAemmAndroid/
aemm platform add https://github.com/mttmllns/aemm-android.git#newBranch
```

This would simplify development and testing of changes to the aemm-android and aemm-ios platform repos.

### PR DESCRIPTION
**Pull request by adzellman**
- Fixes #26
 - Fixes #33
 - Update help documents
 - Combine common platform code

@mttmllns 👍 
@EonerAemm 👍 
@rissmanadobe 👍

### CODE REVIEW

File: src/platform.js [hunk 1]
```diff
@@ -18,12 +18,20 @@
...
  */
+var FS = require('q-io/fs');
+var fs = require('fs');
 var Q = require('q');
 var platformRequire = require('../utils/platformRequire');
 var path = require("path");
-var project = require('./project');
+var rp = require('request-promise');
+var semver = require('semver');
+var _ = require('underscore');
```
Comment (mttmllns) [inline src/platform.js]: :neckbeard:

File: src/platform.js [hunk 2]
```diff
@@ -40,19 +48,89 @@ function install(options, platform)
...
+            // We should have an array of length 1 because we should have only installed one platform during this operation.
+            platform = _.first(_.difference(newPlatformsList, installedPlatformsList));
+        }
+
+        if (platform in platforms){
+            var platformModule = platformRequire("platform", platform);
+            return platformModule.post_add();
+        }
+        else {
+            events.emit("info", "Custom platform provided. Skipping post_add. This may result in missing some aemm plugins.");
```
Comment (mttmllns) [inline src/platform.js]: Why skip this step?

Comment (adzellman) [inline src/platform.js]: I guess we don't have to. In development I did this before I did the stuff related to L125, so there was no "platform" when it was provided by filepath.

Comment (mttmllns) [inline src/platform.js]: Ah, that makes sense.  Do forks of aemm-android and aemm-ios count as known platforms "android" and "ios"?

Comment (adzellman) [inline src/platform.js]: https://github.com/apache/cordova-lib/blob/master/cordova-lib/src/cordova/platform.js#L318

This is where the `cordova platform add` command–which we call into–determines the platform of a repository. It parses it out of the name in package.json. If the forked repository has not modified this, then we're safe.

Comment (mttmllns) [inline src/platform.js]: Cool.  This PR looks good then, unless you want to modify this comment.

**Comment by EonerAemm** [comment]
Getting this error when using platform add 
`aemm platform add android -d`

```
Error: TypeError: response.forEach is not a function
    at rp.then (/Users/eoner/git-aemmobile/src/platform.js:163:18)
    at tryCatcher (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:581:18)
    at Promise._settlePromises (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:697:14)
    at Async._drainQueue (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues (/Users/eoner/git-aemmobile/node_modules/request-promise/node_modules/bluebird/js/main/async.js:15:14)
    at runCallback (timers.js:574:20)
    at tryOnImmediate (timers.js:554:5)
    at processImmediate [as _immediateCallback] (timers.js:533:5)
```

**Pull request merged**